src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyDiscoveryManager.cs

```diff
@@ -130,25 +131,38 @@ public void Initialize(bool skipDefaultAdapters)
         /// <inheritdoc/>
         public void DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEventsHandler2 eventHandler)
         {
+
             if (this.proxyOperationManager == null)
             {
                 this.proxyOperationManager = this.proxyOperationManagerCreator(
                     discoveryCriteria.Sources.First(),
                     this);
-
+                //TODO: here or below wrap the testhost manager to return sources that use the correct remote path
                 this.testHostManager = this.proxyOperationManager.TestHostManager;
                 this.requestData = this.proxyOperationManager.RequestData;
             }
 
-            this.baseTestDiscoveryEventsHandler = eventHandler;
+            if (this.testHostManager.GetType().Name.Contains("Uwp"))
```

Comment:
Ignore all this, it just allows me to special case this for uwp, and change the responses.